src/lib.rs

```diff
@@ -1,38 +1,286 @@
-state of the player character, like `OnFoot` or `InCar`.
+## Prelude
 
-For example, if you have a player character that can be on foot or drive a car, you can swap the context to have the same keys
-perform different actions. On foot, pressing Space makes the character jump, while when driving, the same key acts as a brake.
+We provide a [`prelude`] module, which exports most of the typically used traits and types.
 
-Entities can have any number of contexts, with evaluation order controlled by a defined priority. Actions can also consume inputs,
-allowing you to layer behaviors on top of each other.
+## Plugins
 
-Instead of reacting to raw input data like "Released" or "Pressed", the crate provides modifiers and conditions.
+Add [`EnhancedInputPlugin`] to your app:
 
-[`Modifiers`](input_modifier) let you change the input before passing it to the action. We provide common modifiers,
-like [`DeadZone`], [`Negate`], etc., but you can add your own by implementing [`InputModifier`].
+```
+use bevy::prelude::*;
+use bevy_enhanced_input::prelude::*;
 
-[`Conditions`](input_condition) define how an action activates. We also provide built-in conditions, such as [`Press`],
-[`Release`], [`Hold`], etc. You can also add your own by implementing [`InputCondition`].
+let mut app = App::new();
+app.add_plugins((MinimalPlugins, EnhancedInputPlugin));
+```
 
-# Quick start
+## Defining actions
 
-We provide a [`prelude`] module, which exports most of the typically used traits and types.
+Actions represented by unit structs that implement the [`InputAction`] trait, which also requires
+[`Debug`] which we use for logging. You can use the provided derive macro for this.
+
+```
+# use bevy::prelude::*;
+# use bevy_enhanced_input::prelude::*;
+#[derive(Debug, InputAction)]
+#[input_action(output = bool)]
+struct Jump;
+
+#[derive(Debug, InputAction)]
+#[input_action(output = Vec2)]
+struct Move;
+```
+
+The `output` defines the [`InputAction::Output`] type and is the only required parameter. See the [`InputAction`]
+documentation for all parameters.
+
+## Defining input contexts
+
+Actions are bound to input contexts. Depending on your type of game, you may have a single global context or multiple
+contexts for different gameplay states. Consider contexts as sets of gameplay actions.
+
+Input contexts are represented by unit structs that implement the [`InputContext`] trait. You can use the provided derive
+macro for this. Unlike actions, all contexts need to be registered in the app using [`InputContextAppExt::add_input_context`].
+
+```
+# use bevy::prelude::*;
+# use bevy_enhanced_input::prelude::*;
+# let mut app = App::new();
+# app.add_plugins(EnhancedInputPlugin);
+app.add_input_context::<OnFoot>();
+
+#[derive(InputContext)]
+struct OnFoot;
+```
+
+No required parameters needed. See the [`InputContext`] documentation for all parameters.
```

Comment:
This would be more helpfully framed as "you can provide additional parameters..."